services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,18 +45,71 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+  if (event == NULL)
+  {
+    return ERR_CODE_INVALID_ARG; 
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, (void *) event, 0) == pdTRUE)
+  {
+    return ERR_CODE_SUCCESS;
+  }  
+
+  return ERR_CODE_QUEUE_FULL; 
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = OVER_TEMPERATURE_SHUTDOWN;
+  thermalMgrSendEvent(&interrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  /* Implement this task */   
+  lm75bd_config_t data = *(lm75bd_config_t *) pvParameters;
```

Comment:
In general, the greater the variable scope, the more descriptive the variable name should be. So make this `lm75bdConfig` or something.